functable.c

```diff
@@ -56,27 +56,24 @@ static void init_functable(void) {
     // Select arch-optimized functions
 
     // X86 - SSE2
-#ifdef X86_SSE2
+#if defined(X86_SSE2) || defined(X86_SSE2_CHUNKSET)
 #  if !defined(__x86_64__) && !defined(_M_X64) && !defined(X86_NOCHECK_SSE2)
     if (x86_cpu_has_sse2)
 #  endif
+    {
+#  ifdef X86_SSE2
         ft.slide_hash = &slide_hash_sse2;
-#endif
-#if defined(X86_SSE2) && defined(HAVE_BUILTIN_CTZ)
-    if (x86_cpu_has_sse2) {
+#    ifdef HAVE_BUILTIN_CTZ
         ft.longest_match = &longest_match_sse2;
         ft.longest_match_slow = &longest_match_slow_sse2;
         ft.compare256 = &compare256_sse2;
-    }
-#endif
-#ifdef X86_SSE2_CHUNKSET
-# if !defined(__x86_64__) && !defined(_M_X64) && !defined(X86_NOCHECK_SSE2)
-    if (x86_cpu_has_sse2)
-# endif
-    {
+#    endif
+#  endif
+#  ifdef X86_SSE2_CHUNKSET
         ft.chunksize = &chunksize_sse2;
```

Comment:
`chunksize_sse2` should use `X86_SSE2`, not `X86_SSE2_CHUNKSET`, otherwise the SSE 4.1 variant could possibly use wrong variant of `chunksize`. This also allows leaving out SSE 2 variants when target processor supports SSE 4.1.

Reply:
This PR did not change that, so that does not look like a regression.
But it looks like what you describe would get fixed by #1424.

Reply:
I know... When that was introduced, I noticed the possible issue... But back then we had no plans to explicitly eliminate out code paths for older instruction sets to reduce code size and dead code. We pretty much relied on compiler doing that implicitly.